### ISSUE TITLE
Recovering plane orients it vertically

### ISSUE THREAD

**Issue by ianjirka** [magico13/KCT#69]
If you recover a plane using the KCT 'recover active vehicle' button, it puts it back into inventory pointed nose-upward.
If you re-launch it, it will be oriented on the runway like a rocket. :-)

The workaround is to edit it in the SPH.  It will be vertically oriented, but clicking on the command pod will instantly reorient it correctly.  You can then save and roll it out.

KSP 1.0.2, KCT 1.1.6

**Comment by magico13** [comment]
Thanks for the report. I know about this, but didn't want to hold off on the release that added the recovery feature any longer. I'm not 100% sure if I'll be able to fix it, especially because it seems like KSP knows there orientation is messed up (so if I manually tell it to rotate 90 degrees it might flip  another 90 if you click the command pod).

I've got finals the next two weeks, but afterward I might have enough free time to take a look at this while I work on other areas of the code.

**Comment by ianjirka** [comment]
No worries.  The workaround isn't too bad. 
BTW - I noticed it says 'Recovering to VAB' instead of SBH - maybe that is a clue.

**Comment by Rodhern** [comment]
I can't remember if this observation has been mentioned before. Today, in the flight scene, when I pressed recover to SPH, in one frame before the scene changed I saw my plane rotated 90 degrees. It was just one frame, so I can't say really how the plane looked, but it surprised me in the sense that until now I thought the rotation bug happened sometime after the recovery, but quite possibly it happens just a fraction of a second before the recovery (?).

**Comment by magico13** [comment]
> in one frame before the scene changed I saw my plane rotated 90 degrees.

At some point in the past I changed the code so that it makes a backup of the vessel and then loads that backup, checking for any exceptions and if any occur then it doesn't recover it. What you're seeing is probably that load of the vessel. Because it's saved with the wrong orientation, when it gets loaded to check for problems it'll have the same wrong one. The bug is likely during the save process when trying to normalize the position and rotation back to "default".

**Comment by tivec** [comment]
I'd like to chime in and say this just happened for me on an aircraft in version v1.4.0.69 on KSP 1.4.

**Comment by nikolain** [comment]
Yes, I'm seeing vertically oriented aircraft as well on recovery. This is on KSP 1.3.1 + RP-1.

Simple rotating it back in Edit mode is not a good workaround, since somehow Edit percentage get's calculated to 6% or so, that requires weeks of SPH work.
